codalab/worker/docker_utils.py

```diff
@@ -261,6 +265,55 @@ def get_container_stats(container):
     return stats
 
 
+@wrap_exception('Unable to check Docker API for container')
+def get_container_stats_on_mac(container: docker.models.containers.Container):
+    if container_exists(container):
+        return get_container_stats_helper_by_name(container.name)
+    else:
+        return 0.0, 0
+
+
+@wrap_exception('Unable to check Docker API for container')
+def get_container_stats_helper_by_name(container_name: str) -> Tuple[float, int]:
+    try:
+        container_stats: dict = client.containers.get(container_name).stats(stream=False)
+
+        cpu_usage: float = get_cpu_usage(container_stats)
+        memory_limit: int = get_memory_limit(container_stats)
+
+        return cpu_usage, memory_limit
+    except docker.errors.NotFound:
+        raise
+
+
+# The ref of the way of calculation:
+# https://www.jcham.com/2016/02/09/calculating-cpu-percent-and-memory-percentage-for-containers/
+def get_cpu_usage(container_stats: dict) -> float:
```

Comment:
Add a docstring here describing what's being calculated.

We should also link, in addition to the above link, to this link (which shows how the actual Docker client uses this same code): https://github.com/moby/moby/blob/131e2bf12b2e1b3ee31b628a501f96bbb901f479/api/client/stats.go#L309